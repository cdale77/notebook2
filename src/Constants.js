const Constants = {
  DEFAULT_STATE: {
    session: {
      signedIn: false,
      uuid: ""
    },
    flash: {
      flashType: "",
      message: ""
    }
  },
  ACTIONS: {
    REQUEST_START: "REQUEST_START",
    REQUEST_END: "REQUEST_END",
    FLASH_SUCCESS: "FLASH_SUCCESS",
    FLASH_ERROR: "FLASH_ERROR",
    FLASH_CLEAR: "FLASH_CLEAR",
    NEW_SESSION_SUCCESS: "NEW_SESSION_SUCCESS",
    NEW_SESSION_FAILURE: "NEW_SESSION_FAILURE",
    DESTROY_SESSION: "DESTROY_SESSION"
  }
};

export default Constants;
