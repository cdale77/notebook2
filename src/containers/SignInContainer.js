import React from "react";
import { connect } from "react-redux";
//import SignInForm from "./../components/forms/SignInForm";
//import SessionThunks from "./../thunks/SessionThunks";

const mapStateToProps = state => {
  return {};
};

const mapDispatchToProps = dispatch => {
  return {
    signIn: form => {
      console.log("sign in");
      //     dispatch(SessionThunks.signIn(form.email, form.password));
    }
  };
};

class SignInContainer extends React.Component {
  render() {
    return (
      <div className="sign-in-container">
        <h2>Sign In</h2>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SignInContainer);
