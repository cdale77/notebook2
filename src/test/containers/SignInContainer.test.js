import React from "react";
import Renderer from "react-test-renderer";
import { Provider } from "react-redux";
import SignInContainer from "../../containers/SignInContainer";
import configureMockStore from "redux-mock-store";
import Constants from "../../Constants";

describe("SignInContainer", () => {
  const store = configureMockStore()(Constants.DEFAULT_STATE);
  describe("rendering", () => {
    it("should render without error", () => {
      const tree = Renderer.create(
        <Provider store={store}>
          <SignInContainer />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
