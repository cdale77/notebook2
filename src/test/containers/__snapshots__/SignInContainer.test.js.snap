// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInContainer rendering should render without error 1`] = `
<div
  className="sign-in-container"
>
  <div
    className="sign-in-form"
  >
    <h2>
      Sign In
    </h2>
    <div
      className="form fieldset sign-in-form"
    >
      <form>
        <div
          className="form-group"
        >
          <input
            className="input-email"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="input-password"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="button"
            disabled={false}
            id="submit"
            name="submit"
            onClick={[Function]}
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
